cvmfs/sync_union.h

```diff
@@ -233,12 +252,12 @@ class SyncUnionOverlayfs : public SyncUnion {
                            const std::string &filename);
   void ProcessCharacterDevice(const std::string &parent_dir,
                               const std::string &filename);
-  std::string UnwindWhiteoutFilename(const std::string &filename) const;
+  std::string UnwindWhiteoutFilename(const SyncItem &entry) const;
   std::set<std::string> GetIgnoreFilenames() const;
-  virtual void ProcessFile(SyncItem *entry);
+  virtual void ProcessFile(SyncItem &entry);
```

Comment:
Is this a actually a `const SyncItem &entry`?  Otherwise the pointer version should stay to indicate that `entry` can be changed inside `ProcessFile`.